internal/services/compute/image_resource.go

```diff
@@ -434,15 +469,22 @@ func flattenImageOSDisk(input *images.ImageStorageProfile) []interface{} {
 			if set := v.DiskEncryptionSet; set != nil && set.Id != nil {
 				diskEncryptionSetId = *set.Id
 			}
-			output = append(output, map[string]interface{}{
+
+			properties := map[string]interface{}{
 				"blob_uri":               blobUri,
 				"caching":                caching,
 				"managed_disk_id":        managedDiskId,
 				"os_type":                string(v.OsType),
 				"os_state":               string(v.OsState),
 				"size_gb":                diskSizeGB,
 				"disk_encryption_set_id": diskEncryptionSetId,
-			})
+			}
+
+			if v.StorageAccountType != nil && features.FourPointOhBeta() {
+				properties["storage_type"] = string(*v.StorageAccountType)
+			}
```

Comment:
We should actually be setting a value for this even when the API doesn't return one.

```suggestion
			if features.FourPointOhBeta() {
				storageType := ""
				if v.StorageAccountType != nil {
					storageType = string(*v.StorageAccountType)
				}
				properties["storage_type"] = storageType
			}
```